### PR TITLE
Swift 2.0

### LINKED ISSUE #33: Propagate File and Line Number with Test
XCTAssert is landing inside of us instead of inside of the test.

### PR DESCRIPTION
**Pull request by CodaFi**
Fixes #33.

:warning:  Breaking Changes :warning: 

This will be released as SwiftCheck 0.3.0.

### CODE REVIEW

File: SwiftCheck/Check.swift [hunk 1]
```diff
@@ -8,47 +8,61 @@
...
+	let line : UInt
+
+	private init(msg : String, file : String, line : UInt) {
+		self.msg = msg
+		self.file = file
+		self.line = line
+	}
+}
+
+public func +=(checker : AssertiveQuickCheck, test : Testable) {
```
Comment (CodaFi) [inline SwiftCheck/Check.swift]: This was the compromise I made (the closest thing to `=` I was comfortable using).

**Comment by CodaFi** [comment]
Other ideas (because `+=` is kinda sketch):

`<=>`: Bi-implication
`|=`: Turnstile
`|=|`: Fence
`<~`: Tail

**Comment by CodaFi** [comment]
Right now I'm leaning towards a turnstile, because the pun appeals to the mathematician in me.

```swift
property("I know this to be true") |= forAll {
 ///
}
```

**Comment by CodaFi** [comment]
Roping in Twitter people:
@griotspeak
@robrix 
@pthariensflame

**Comment by griotspeak** [comment]
Each of those seem preferable to `+=`. You mentioned was of typing. `~~` seems pretty easy to type as well and I am unaware of a different meaning for it.

**Comment by pthariensflame** [comment]
If you're going to go with a turnstile, why not a proper one, and pointing in the proper direction (*i.e.*, `-|`)?  On the other hand, I'm not sure that a turnstile, of any form or appearance, is a good idea in the first place.  If `:=` we're valid, I would suggest that.  :(

**Comment by CodaFi** [comment]
@pthariensflame it doesn't work out for plain old vdash but double-dash makes entailment (if you view the property() part as Γ).

**Comment by CodaFi** [comment]
But fine, I definitely could see how that causes headaches in the future.

What about something as simple as `>>`?

**Comment by griotspeak** [comment]
Let us not rehash the mistakes of C++. Let the bit shift operators be.

**Comment by CodaFi** [comment]
RAC has dibs on `<~`, so that's kinda out.

😕

**Comment by CodaFi** [comment]
But Swiftz had dibs on `<-` and now that's been removed so it's up for grabs.

**Comment by pthariensflame** [comment]
:+1: for `<-`; we might be even be able to reuse it elsewhere.

**Comment by CodaFi** [comment]
Going to push that for now.  There's no rush with this.  I want to try to time 0.3.0 for this and Swiftz with the next release of Swift next week at WWDC.

**Comment by robrix** [comment]
Agreed, `<-` is pretty nice.

**Comment by CodaFi** [comment]
To merge and branch later or close and branch now...

**Comment by robrix** [comment]
I’m very much looking forward to a Swift 2-compatible SwiftCheck either way :smile:

**Comment by CodaFi** [comment]
If only I could get this to stop tripping asserts in Swiftc :disappointed:

File: SwiftCheck/Gen.swift [hunk 2]
```diff
@@ -19,9 +23,99 @@ public struct Gen<A> {
...
+			return xs[i]
+		}
+	}
+
+	/// Constructs a Generator that uses a given array to produce smaller arrays composed of its 
+	/// initial segments.  The size of each initial segment increases with the receiver's size 
+	/// parameter.
+	///
+	/// The input array is required to be non-empty.
+	public static func fromInitialSegmentsOf(xs : [A]) -> Gen<A> {
```
Comment (CodaFi) [inline SwiftCheck/Gen.swift]: These `from*` functions feel out of place with the other constructors, but I couldn't quite think of a good verb.  Ideas?

**Comment by CodaFi** [comment]
:boat:

**Pull request merged**